### DISCUSSION TITLE
kiosk-mode on iPad freezes

### DISCUSSION THREAD

**Discussion by hajo62** [NemesisRE/kiosk-mode#186] [General]
Someone there who runs the **kiosk-mode on an iPad** and has some **graphical elements** (like apex charts) on the dashboard?

When I put a graphical element on the dashboard it does stop to update the shown values in all cards.

**Comment by NemesisRE** [comment]
With kiosk-mode disabled it works?

**Comment by hajo62** [comment]
> With kiosk-mode disabled it works?
Not sure. Tried a quick test, but with that both _modes_ worked today

So I assume, that the freeze occurs not directly, but after a while?! - Maybe when the iPad goes to kind of sleep and not only darkens/locks the screen?

What I do and observe:

- Display a dashboard (in kiosk mode or without)
- After a while, the display is automatically switched off
- I switch on the display by sending a push notification

**Reply by NemesisRE** [reply]
I would assume that the problem is not related to kiosk-mode but rather to sleep/hibernation of the ipad

I think I had the same issue with an android tablet when I used the home assistant app I "fixed" it by using the fully kiosk browser app which reloaded the page after coming out of sleep.

My guess it that the js processes get stopped and not restarted without reload.

**Reply by elchininet** [reply]
@hajo62,

Make sure to switch off this option in the settings (it is on by default):

![Screenshot_2024-01-17-13-01-07-87_c3a231c25ed346e59462e84656a70e50.jpg](https://github.com/NemesisRE/kiosk-mode/assets/3728220/04ea3a9c-586f-4e2c-9844-629713d8a289)

**Comment by hajo62** [comment]
Hhm,
@NemesisRE : I use a kiosk browser on the iPad, too.

@elchininet : To be honest, (I'm newbie to kiosk_mode stuff) I don't know, where to find that "don't close connection"-setting.

**Reply by elchininet** [reply]
@hajo62,

That option is not from `kiosk-mode` but from Home Assistant. Click on your profile avatar and scroll until you find it.

**Reply by hajo62** [reply]
Thx. Switched that off right now.
